ReactiveCocoa/Swift/Action.swift

```diff
@@ -127,6 +127,101 @@ public final class Action<Input, Output, Error: ErrorType> {
 	}
 }
 
+/// Wraps an Action for use by a GUI control (such as `NSControl` or
+/// `UIControl`), with KVO, or with Cocoa Bindings.
+public final class CocoaAction: NSObject {
+	/// The selector that a caller should invoke upon a CocoaAction in order to
+	/// execute it.
+	public class var selector: Selector {
+		return "execute:"
+	}
+
+	/// Whether the action is enabled.
+	///
+	/// This property will only change on the main thread, and will generate a
+	/// KVO notification for every change.
+	public var enabled: Bool {
+		return _enabled
+	}
+
+	/// Whether the action is executing.
+	///
+	/// This property will only change on the main thread, and will generate a
+	/// KVO notification for every change.
+	public var executing: Bool {
+		return _executing
+	}
+
+	private var _enabled = false
+	private var _executing = false
+	private let _execute: AnyObject? -> ()
+	private let disposable = CompositeDisposable()
+
+	/// Initializes a Cocoa action that will invoke the given Action by
+	/// transforming the object given to execute().
+	public init<Input, Output, Error>(_ action: Action<Input, Output, Error>, _ inputTransform: AnyObject? -> Input) {
+		_execute = { input in
+			let producer = action.apply(inputTransform(input))
+			producer.start()
+		}
+
+		super.init()
+
+		let enabledDisposable = action.enabled.producer
+			|> observeOn(UIScheduler())
+			|> start(Event.sink(next: { [weak self] value in
```

Comment:
These explicit uses of `Event.sink()` seemed to be necessary to work around a compiler crash. :disappointed: